crates/agglayer-node/src/kernel/mod.rs

```diff
@@ -343,6 +352,9 @@ where
             warn!(hash, "Transaction already settled: {:?}", tx);
         }
 
+        self.rate_limiter
+            .limit_send_tx_now(signed_tx.tx.rollup_id)?;
```

Comment:
If the transaction fails, the rate limit is still applied right?

Reply:
> If the transaction fails, the rate limit is still applied right?

Yea, if the tx submission to L1 fails, it still contributes to the rate limit in this case. It would be more accurate to use the actual L1 state to determine how many transaction settlements actually happened. That would probably be a fair bit more complicated. Especially considering the various delays and weird "in-between" states involved. For example, should a settlement transaction in mempool contribute towards rate limit? At that point, whether it is going to be confirmed or not is outside of our control to a large extent.